datafusion/src/physical_plan/expressions/min_max.rs

```diff
@@ -38,6 +38,15 @@ use arrow::{
 
 use super::format_state_name;
 
+// min/max aggregation only returns a value for each group
```

Comment:
Maybe we could modify the comment here to explain that since the `Min/Max` aggregate can handle `Dictionary` encoded inputs, but always produces "unpacked" (aka non `Dictionary`) output we need to adjust the output data type to reflect this.